cli/src/pcluster/config/cluster_config.py

```diff
@@ -2324,7 +2312,19 @@ def _instance_type_info(self) -> InstanceTypeInfo:
     @property
     def disable_simultaneous_multithreading_manually(self) -> bool:
         """Return true if simultaneous multithreading must be disabled with a cookbook script."""
-        return self.disable_simultaneous_multithreading and self.instance_type_info.default_threads_per_core() > 1
+        return self.disable_simultaneous_multithreading and self._instance_type_info.default_threads_per_core() > 1
+
+    def _instance_id_from_capacity_target(self):
+        """Return the instance id from the configured CapacityReservationId, if any."""
+        instance_type = None
+        capacity_reservation_id = (
+            self.capacity_reservation_target.capacity_reservation_id if self.capacity_reservation_target else None
+        )
+        if capacity_reservation_id:
+            capacity_reservations = AWSApi.instance().ec2.describe_capacity_reservations([capacity_reservation_id])
```

Comment:
is this call cached?

Reply:
Short answer: yes,

Long answer: for this call there is an internal caching mechanism (unit tested), that was already in place.

```
        missed_capacity_reservations = []
        for capacity_reservation_id in capacity_reservation_ids:
            cached_data = self.capacity_reservations_cache.get(capacity_reservation_id)
            if cached_data:
                result.append(cached_data)
            else:
                missed_capacity_reservations.append(capacity_reservation_id)
        if missed_capacity_reservations:
            response = list(
                self._paginate_results(
                    self._client.describe_capacity_reservations, CapacityReservationIds=missed_capacity_reservations
                )
            )
```
this because there are multiple `describe` calls with different capacity_reservation_ids.